src/C4Group.cpp

```diff
@@ -670,6 +670,35 @@ bool C4Group::Open(const char *szGroupName, bool fCreate)
 		// Open group or folder
 		return OpenReal(szGroupNameN);
 	}
+#ifdef __linux__
+	else
+	// check if file exists in user data directory
+    {
+        StdStrBuf data_directory(getenv("XDG_DATA_HOME"));
+	    if (!data_directory.getSize()) {
+	        data_directory = getenv("HOME");
+	        data_directory += "/.local/share";
+	    }
+	    data_directory += "/legacyclonk/";
+	    StdStrBuf filename = data_directory + szGroupNameN;
+	    if (FileExists(filename.getData()))
+        {
+	        Init();
+	        return OpenReal(filename.getData());
+        }
+	    // check if file exists in system distribution directory
+	    filename = "/usr/share/legacyclonk/";
+	    filename += szGroupNameN;
+	    if (FileExists(filename.getData()))
+        {
+	        Init();
+	        return OpenReal(filename.getData());
+        }
+
+	}
+#endif
+
+
```

Comment:
Sorry, no. C4Group isn't even remotely the right place for this. C4Group should handle file IO, as it already does - it *must not* deal with which operating system or configuration wants what file to be where. Just no. This is also a way to make C4Group access files regardless of what path is actually required. The more I think about it, the worse it gets. Please don't do that.

Also, you hardcoded legacyclonk - the config does this too (which is equally bad, but not your fault).